### PR TITLE
Add vm endpoints

### PR DESCRIPTION
**Pull request by fradinni**

### CODE REVIEW

File: src/sdk/model/vm/guest-customization/guest-customization-update-request.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  constructor(guestCustomizationJson: GuestCustomizationJson);
+  constructor(firstParam: GuestCustomization | GuestCustomizationJson) {
+    if (firstParam instanceof GuestCustomization) {
+      super(firstParam);
+    } else {
+      super(firstParam as GuestCustomizationJson);
+    }
+  }
+
+  setEnabled(enabled: boolean) {
```
Comment (JoMas971) [inline src/sdk/model/vm/guest-customization/guest-customization-update-request.ts]: Maybe we could add some doc for those setters ?

File: src/sdk/model/vm/guest-customization/guest-customization.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    if (firstParam instanceof GuestCustomization) {
+      // Copy constructor
+      this._json = (firstParam as GuestCustomization).json;
+    } else {
+      // Json or empty constructor
+      this._json = (firstParam || {}) as GuestCustomizationJson;
+    }
+  }
+
+  get enabled(): boolean {
```
Comment (JoMas971) [inline src/sdk/model/vm/guest-customization/guest-customization.ts]: Maybe we can add some doc for those getters ?

File: src/sdk/model/vm/guest-customization/guest-customization.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    return this._json.enabled;
+  }
+
+  get changeSid(): boolean {
```
Comment (JoMas971) [inline src/sdk/model/vm/guest-customization/guest-customization.ts]: In the case that this._json is empty, this getter won't return a boolean. You need to handle that case too i guess. Same will apply to other getters too...

File: src/sdk/model/vm/guest-tools/guest-tools.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * VM Guest Tools
+ */
+export class GuestTools {
+  private _json: GuestToolsJson;
+
+  constructor(guestTools: GuestTools);
+  constructor(guestToolsJson: GuestToolsJson);
+  constructor(firstParam: boolean | GuestTools | GuestToolsJson) {
```
Comment (JoMas971) [inline src/sdk/model/vm/guest-tools/guest-tools.ts]: When would we have a boolean as firstParam there ? I can't think of any cases...

File: src/sdk/model/vm/guest-tools/guest-tools.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    if (firstParam instanceof GuestTools) {
+      // Copy constructor
+      this._json = (firstParam as GuestTools).json;
+    } else {
+      // Json or empty constructor
+      this._json = (firstParam || {}) as GuestToolsJson;
```
Comment (JoMas971) [inline src/sdk/model/vm/guest-tools/guest-tools.ts]: And if firstParam is a boolean ? That will fail.

File: src/sdk/model/vm/guest-tools/guest-tools.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    }
+  }
+
+  /**
+   * Returns guest tools status
+   * @returns {string}
+   */
+  get status(): string {
```
Comment (JoMas971) [inline src/sdk/model/vm/guest-tools/guest-tools.ts]: Same here, if this._json is empty, this method won't return a string, but undefined instead.

File: src/sdk/model/vm/summary/vm-summary.ts [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+      // Json or empty constructor
+      this._json = (firstParam || {}) as VmSummaryJson;
+    }
+  }
+
+  /**
+   * Returns reserved CPU.
+   * @returns {number}
+   */
+  get reservedCpu(): number {
```
Comment (JoMas971) [inline src/sdk/model/vm/summary/vm-summary.ts]: Same there. If this._json is empty. this method won't return a number. Same for all other methods.

File: src/sdk/model/vm/guest-customization/guest-customization-update-request.ts [hunk 8]
```diff
@@ -0,0 +1,83 @@
+import { GuestCustomization } from './guest-customization';
+import { GuestCustomizationJson } from './__json__';
+
+export class GuestCustomizationUpdateRequest extends GuestCustomization {
```
Comment (cfsnyder) [inline src/sdk/model/vm/guest-customization/guest-customization-update-request.ts]: this should not extend GuestCustomization

File: src/sdk/model/vm/guest-customization/guest-customization-update-request.ts [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    } else {
+      super(firstParam as GuestCustomizationJson);
+    }
+  }
+
+  setEnabled(enabled: boolean) {
+    this._json.enabled = enabled;
+  }
+
+  setChangeSid(value: boolean) {
```
Comment (cfsnyder) [inline src/sdk/model/vm/guest-customization/guest-customization-update-request.ts]: use property accessor setters like we do everywhere else

File: src/sdk/model/vm/guest-customization/guest-customization.ts (hunk 2 above)
Comment (cfsnyder) [reply src/sdk/model/vm/guest-customization/guest-customization.ts]: @fradinni can we add docs on this one?

File: src/sdk/model/vm/vm-memory-size-update-request.ts [hunk 10]
```diff
@@ -0,0 +1,35 @@
+import { VmMemorySizeUpdateRequestJson } from './__json__';
+
+export class VmMemorySizeUpdateRequest {
+
+  constructor(private _json: VmMemorySizeUpdateRequestJson) {
```
Comment (cfsnyder) [inline src/sdk/model/vm/vm-memory-size-update-request.ts]: can we use the normal overloaded constructor?

File: src/sdk/model/vm/vm-reconfigure-request.ts [hunk 11]
```diff
@@ -0,0 +1,141 @@
+import { VmReconfigureRequestJson } from './__json__';
+import { VmDiskRequestJson } from './__json__/vm-json';
+import { VmCpuCountUpdateRequest } from './vm-cpu-count-update-request';
+import { VmMemorySizeUpdateRequest } from './vm-memory-size-update-request';
+import { GuestCustomizationUpdateRequest } from './guest-customization/guest-customization-update-request';
+
+export class VmReconfigureRequest {
+
+  constructor(private _json: VmReconfigureRequestJson) {
```
Comment (cfsnyder) [inline src/sdk/model/vm/vm-reconfigure-request.ts]: can we use the overloaded constructor?

**Pull request merged**